File: sourceC# - Visual Studio/Workspaces/ESHOP/EShop/Entities/ProductEntity.cs
using EShop.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EShop.Entities
{
    public class ProductEntity : BaseEntity
    {
        public Guid Id { get; set; }
        public String Code { get; set; }
        public String Unit { get; set; }
        public Guid CategoryId { get; set; }
        public Guid ManufacturerId { get; set; }
        public CategoryEntity CategoryEntity { get; set; }
        public ManufacturerEntity ManufacturerEntity { get; set; }
        public List<DiscountEntity> DiscountEntities { get; set; }
        public List<InventoryEntity> InventoryEntities { get; set; }
        public List<IssueNoteLineEntity> IssueNoteLineEntities { get; set; }
        public List<PackEntity> PackEntities { get; set; }
        public List<ProductPictureEntity> ProductPictureEntities { get; set; }
        public List<ProductValueEntity> ProductValueEntities { get; set; }
        public List<ReceiptNoteLineEntity> ReceiptNoteLineEntities { get; set; }

        public ProductEntity():base() { }

        public ProductEntity(Product Product, params object[] args) :base(Product)
        {
		    foreach(object arg in args)
			{
                if (arg is Category Category)
                    CategoryEntity = new CategoryEntity(Category);				
                if (arg is Manufacturer Manufacturer)
                    ManufacturerEntity = new ManufacturerEntity(Manufacturer);				
                if (arg is ICollection<Discount> Discounts)
                    DiscountEntities = Discounts.Select(model => new DiscountEntity(model)).ToList();				
                if (arg is ICollection<Inventory> Inventories)
                    InventoryEntities = Inventories.Select(model => new InventoryEntity(model, model.WareHouse)).ToList();				
                if (arg is ICollection<IssueNoteLine> IssueNoteLines)
                    IssueNoteLineEntities = IssueNoteLines.Select(model => new IssueNoteLineEntity(model, model.IssueNote)).ToList();				
                if (arg is ICollection<Pack> Packs)
                    PackEntities = Packs.Select(model => new PackEntity(model)).ToList();				
                if (arg is ICollection<ProductPicture> ProductPictures)
                    ProductPictureEntities = ProductPictures.Select(model => new ProductPictureEntity(model)).ToList();				
                if (arg is ICollection<ProductValue> ProductValues)
                    ProductValueEntities = ProductValues.Select(model => new ProductValueEntity(model, model.Attribute, model.Language)).ToList();				
                if (arg is ICollection<ReceiptNoteLine> ReceiptNoteLines)
                    ReceiptNoteLineEntities = ReceiptNoteLines.Select(model => new ReceiptNoteLineEntity(model, model.ReceiptNote)).ToList();				
			}
        }
    }

    public class ProductSearchEntity : FilterEntity
    {
        public Guid? Id { get; set; }
        public String Code { get; set; }
        public String Unit { get; set; }
        public Guid? CategoryId { get; set; }
        public Guid? ManufacturerId { get; set; }
    }
}

File: sourceC# - Visual Studio/Workspaces/ESHOP/EShop/FrontEnd/src/app/Module/MProduct/Product.Entity.ts
import { IEntity } from "../IEntity.Entity";
import { PriceListEntity } from "../MPriceList/PriceList.Entity";
import { InventoryEntity } from "../MInventory/Inventory.Entity";
import { PackEntity } from "../MPack/Pack.Entity";
import { DiscountEntity } from "../MDiscount/Discount.Entity";
import { CategoryEntity } from "../MCategory/Category.Entity";
import { ManufacturerEntity } from "../MManufacturer/Manufacturer.Entity";
import { IssueNoteLineEntity } from "../MIssueNoteLine/IssueNoteLine.Entity";
import { ProductPictureEntity } from "../MProductPicture/ProductPicture.Entity";
import { ProductValueEntity } from "../MProductValue/ProductValue.Entity";
import { ReceiptNoteLineEntity } from "../MReceiptNoteLine/ReceiptNoteLine.Entity";

export class ProductEntity extends IEntity {
    Id: number = null;
    Code: string = null;
    CategoryId: string = null;
  Unit: string = null;
  Picture: string = null;
  ManufacturerId: string = null;
  Cx: number = null;

  CategoryEntity: CategoryEntity = null;
  ManufacturerEntity: ManufacturerEntity = null;
  InventoryEntities: InventoryEntity[] = [];
    PackEntities: PackEntity[] = [];
    DiscountEntities: DiscountEntity[] = [];
  IssueNoteLineEntities: IssueNoteLineEntity[] = [];
  ProductPictureEntities: ProductPictureEntity[] = [];
  ProductValueEntities: ProductValueEntity[] = [];
  ReceiptNoteLineEntities: ReceiptNoteLineEntity[] = [];

    public constructor() {
        super();
    }

}
export class ExtraEntity {
    Id: number = null;
    Code: string = null;
    Name: string = null;
    Value: string = null;
}

